components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -110,6 +110,10 @@ export function CreateWorkspacePage() {
     const workspaceContext = useWorkspaceContext(contextURL);
     const needsGitAuthorization = useNeedsGitAuthorization();
 
+    useEffect(() => {
+        setContextURL(StartWorkspaceOptions.parseContextUrl(location.hash));
```

Comment:
How about update dependencies of `handleContextURLChange`? So that changes like search params could affect UI too

Reply:
@mustard-mh not sure if I follow? `handleContextURLChange` seems pretty tailor fit for the dropdown behavior, so I don't think we should use it here.